_1327/main/tests.py

```diff
@@ -522,3 +523,103 @@ def test_remind_users_about_due_unpublished_minutes_documents(self):
+		cls.poll = mommy.make(Poll, text=text2, title="TestPoll")
+		cls.information_document = mommy.make(InformationDocument, text=text3, title="TestInformationDocument")
+		cls.minutes_document_w_script = mommy.make(MinutesDocument, text=text4, title="TestMinuteWithScript")
+		cls.information_document_never = mommy.make(InformationDocument, text=text5, title="TestInformationDocumentNever")
+		cls.group = mommy.make(Group)
+		cls.minutes_document.set_all_permissions(cls.group)
+		cls.poll.set_all_permissions(cls.group)
+		cls.information_document.set_all_permissions(cls.group)
+		cls.minutes_document_w_script.set_all_permissions(cls.group)
+
+	def test_all_types_of_documents(self):
+		search_string = "all three"
+
+		response = self.app.get(reverse('index'), user=self.user)
+
+		form = response.forms["general_search"]
+		form.set('search_phrase', search_string)
+
+		response = form.submit()
+
+		self.assertIn('TestMinute', response)
+		self.assertIn('TestPoll', response)
+		self.assertIn('TestInformationDocument', response)
+		self.assertNotIn('TestInformationDocumentNever', response)
+
+	def test_case_insensitive_result(self):
+		search_string = "case"
+
+		response = self.app.get(reverse('index'), user=self.user)
+
+		form = response.forms["general_search"]
+		form.set('search_phrase', search_string)
+
+		response = form.submit()
+
+		self.assertIn('TestMinute', response)
+		self.assertNotIn('TestPoll', response)
+		self.assertNotIn('TestInformationDocument', response)
+		self.assertNotIn('TestInformationDocumentNever', response)
+
+		self.assertIn('<b>Case</b> notB notO', response)
+
+	def test_substring_result(self):
+		search_string = "bstrin"
+
+		response = self.app.get(reverse('index'), user=self.user)
+
+		form = response.forms["general_search"]
+		form.set('search_phrase', search_string)
+
+		response = form.submit()
+
+		self.assertIn('TestPoll', response)
+		self.assertNotIn('TestMinute', response)
+		self.assertNotIn('TestInformationDocument', response)
+		self.assertNotIn('TestInformationDocumentNever', response)
+
+		self.assertIn('su<b>bstrin</b>g notB notO', response)
+
+	def test_nothing_found_message(self):
+		search_string = "not in the minutes"
+
+		response = self.app.get(reverse('index'), user=self.user)
+
+		form = response.forms["general_search"]
+		form.set('search_phrase', search_string)
+
+		response = form.submit()
+
+		self.assertIn('No documents containing "not in the minutes" found.', response.body.decode('utf-8'))
+		self.assertNotIn('notB', response)
+		self.assertNotIn('notO', response)
+
+	def test_correct_escaping(self):
+		search_string = "<script>alert(Hello);</script>"
+
+		response = self.app.get(reverse('index'), user=self.user)
+
+		form = response.forms["general_search"]
+		form.set('search_phrase', search_string)
+
+		response = form.submit()
+
+		self.assertIn('<b>&lt;script&gt;alert(Hello);&lt;/script&gt;</b> something else', response.body.decode('utf-8'))
```

Comment:
How about a test that tests, whether a user is actually allowed to search/see the document?